node/src/components/consensus/highway_core/finality_detector.rs

```diff
@@ -82,21 +91,25 @@ impl<C: Context> FinalityDetector<C> {
 
     /// Returns the next block, if any has been finalized since the last call.
     pub(super) fn next_finalized<'a>(&mut self, state: &'a State<C>) -> Option<&'a C::Hash> {
+        let start_time = Timestamp::now();
         let candidate = self.next_candidate(state)?;
         // For `lvl` → ∞, the quorum converges to a fixed value. After level 63, it is closer
         // to that limit than 1/2^-63. This won't make a difference in practice, so there is no
         // point looking for higher summits.
         let mut target_lvl = 63;
         while target_lvl > 0 {
+            trace!("looking for summit with level {}", target_lvl);
             let lvl = self.find_summit(target_lvl, candidate, state);
             if lvl == target_lvl {
                 self.last_finalized = Some(candidate.clone());
+                trace!("found finalized block in {}", start_time.elapsed());
```

Comment:
It's be better to explicitly have a variable that captures `start_time.elapsed()`, I get a little anxious when business logic is contained inside logging calls.